docker/docs/dev/Dockerfile

```diff
@@ -22,9 +22,9 @@ RUN git clone https://github.com/CrayLabs/SmartRedis.git --branch develop --dept
     && python -m pip install . \
     && rm -rf ~/.cache/pip
 
-# address needs of nbsphinx
+# address needs of nbsphinx and ml docs
```

Comment:
@Spartee - Can you give some more context on the ML dependencies in the docs build? Specifically, what ML docs require this?

Reply:
Nevermind, I understand now. I commented them out and saw a bunch of related errors/warnings in the doc build:

```

#13 2.974 [autosummary] generating autosummary for: api/smartredis_api.rst, api/smartsim_api.rst, changelog.rst, code_of_conduct.rst, community.rst, developer.rst, experiment.rst, index.rst, installation.rst, launchers.rst, ..., sr_data_structures.rst, sr_runtime.rst, tutorials/01_getting_started/01_getting_started.ipynb, tutorials/02_using_clients.rst, tutorials/03_lattice_boltz_analysis.rst, tutorials/04_inference.rst, tutorials/05_ml_training/train_tf.ipynb, tutorials/05_ml_training/train_torch.ipynb, tutorials/05_training.rst, tutorials/06_starting_ray/06_starting_ray_builtin.ipynb
#13 3.075 Failed to import smartsim.ml.tf.utils.
#13 3.075 Possible hints:
#13 3.075 * AttributeError: module 'smartsim.ml' has no attribute 'tf'
#13 3.075 * ModuleNotFoundError: TensorFlow 2.5.2 is not installed. Please install it to use smartsim.tf
#13 3.075 parsing fortran sources...{'before': '', 'this': 'use', 'after': ', intrinsic :: iso_fortran_env, only: stderr => error_unit '}

...

#13 17.44 WARNING: autodoc: failed to import module 'tf.utils' from module 'smartsim.ml'; the following exception was raised:
#13 17.44 TensorFlow 2.5.2 is not installed. Please install it to use smartsim.tf
#13 17.44 WARNING: autodoc: failed to import class 'tf.StaticDataGenerator' from module 'smartsim.ml'; the following exception was raised:
#13 17.44 TensorFlow 2.5.2 is not installed. Please install it to use smartsim.tf
#13 17.44 WARNING: autodoc: failed to import class 'tf.DynamicDataGenerator' from module 'smartsim.ml'; the following exception was raised:
#13 17.44 TensorFlow 2.5.2 is not installed. Please install it to use smartsim.tf
#13 17.44 WARNING: autodoc: failed to import class 'torch.StaticDataGenerator' from module 'smartsim.ml'; the following exception was raised:
#13 17.44 No module named 'torch'
#13 17.44 WARNING: autodoc: failed to import class 'torch.DynamicDataGenerator' from module 'smartsim.ml'; the following exception was raised:
#13 17.44 No module named 'torch'
#13 17.44 WARNING: autodoc: failed to import class 'torch.DataLoader' from module 'smartsim.ml'; the following exception was raised:
```